wazo_agid/objects.py

```diff
@@ -906,33 +906,44 @@ def __init__(self, agi, cursor, context):
 class CallerID(object):
     @staticmethod
     def parse(callerid):
+        logger.debug('caller_id parse: parsing "%s"', callerid)
         m = CALLERID_MATCHER(callerid)
 
         if not m:
+            logger.debug('caller_id parse: could not match callerid, giving up')
             return
 
         calleridname = m.group(1)
         calleridnum = m.group(3)
+        logger.debug('caller_id parse: calleridname: "%s", calleridnum: "%s"', calleridname, calleridnum)
 
         if calleridname is None:
             calleridname = m.group(2)
+            logger.debug('caller_id parse: calleridname: "%s", calleridnum: "%s"', calleridname, calleridnum)
 
             if calleridnum is None and CALLERIDNUM_MATCHER(calleridname):
                 calleridnum = m.group(2)
+                logger.debug('caller_id parse: calleridname: "%s", calleridnum: "%s"', calleridname, calleridnum)
 
+        logger.debug('caller_id parse: done')
         return (calleridname, calleridnum)
 
     @staticmethod
     def set(agi, callerid):
+        logger.debug('caller_id set: parsing "%s"', callerid)
         cid_parsed = CallerID.parse(callerid)
+        logger.debug('caller_id set: parsed caller id to "%s"', cid_parsed)
```

Comment:
Can we move the log under the `if not cid_parsed`. it will avoid redundant log if cid_parsed is None/empty